haystack_experimental/components/image_converters/image_utils.py

```diff
@@ -256,40 +208,131 @@ def _convert_pdf_to_pil_images(
-        will convert pages 1, 2, 3, 5, 8, 10, 11, 12.
-    :param size: If provided, resizes the image to fit within the specified dimensions (width, height) while
-        maintaining aspect ratio. This reduces file size, memory usage, and processing time, which is beneficial
-        when working with models that have resolution constraints or when transmitting images to remote services.
     :returns:
-        A list of tuples, each tuple containing the page number and the base64-encoded image string.
+        A list of _ImageSourceInfo dictionaries, each containing the path and type of the image.
+        If the image is a PDF, the dictionary also contains the page number.
+    :raises ValueError: If the document is missing the file_path_meta_field key in its metadata, the file path is
+        invalid, the MIME type is not supported, or the page number is missing for a PDF document.
+    """
+    images_source_info: List[_ImageSourceInfo] = []
+    for doc in documents:
+        file_path = doc.meta.get(file_path_meta_field)
+        if file_path is None:
+            raise ValueError(
+                f"Document with ID '{doc.id}' is missing the '{file_path_meta_field}' key in its metadata."
+                f" Please ensure that the documents you are trying to convert have this key set."
+            )
+
+        resolved_file_path = Path(root_path, file_path)
+        if not resolved_file_path.is_file():
+            raise ValueError(
+                f"Document with ID '{doc.id}' has an invalid file path '{resolved_file_path}'. "
+                f"Please ensure that the documents you are trying to convert have valid file paths."
+            )
+
+        mime_type = doc.meta.get("mime_type") or mimetypes.guess_type(resolved_file_path)[0]
+        if mime_type not in IMAGE_MIME_TYPES:
+            raise ValueError(
+                f"Document with file path '{resolved_file_path}' has an unsupported MIME type '{mime_type}'. "
+                f"Please ensure that the documents you are trying to convert are of the supported "
+                f"types: {', '.join(IMAGE_MIME_TYPES)}."
+            )
+
+        image_info: _ImageSourceInfo = {"path": resolved_file_path, "mime_type": mime_type}
+
+        # If mimetype is PDF we also need the page number to be able to convert the right page
+        if mime_type == "application/pdf":
+            page_number = doc.meta.get("page_number")
+            if page_number is None:
+                raise ValueError(
+                    f"Document with ID '{doc.id}' comes from the PDF file '{resolved_file_path}' but is missing "
+                    f"the 'page_number' key in its metadata. Please ensure that PDF documents you are trying to "
+                    f"convert have this key set."
+                )
+            image_info["page_number"] = page_number
+
+        images_source_info.append(image_info)
+
+    return images_source_info
+
+
+class _PdfPageInfo(TypedDict):
```

Comment:
Small nit, I think we should use capital PDF like we do in our converter names 
```suggestion
class _PDFPageInfo(TypedDict):
```